### ISSUE TITLE
[bitnami/kafka] .targetPod resolves to empty when setting up Name resolution with External-DNS

### ISSUE THREAD

**Issue by eli-gc** [bitnami/charts#19556]
### Name and Version

bitnami/kafka 25.1.10

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Deploy kafka to kubernetes using below defined values.yaml. 

### Are you using any custom parameters or values?

Using this in my values.yaml:

```yaml
externalAccess:
  controller:
    service:
      annotations:
        external-dns.alpha.kubernetes.io/hostname: "{{ .targetPod }}.dev.local"
```

### What is the expected behavior?

If I describe the svc in kubernetes (kubectl describe svc <svc-name>), then I should see the annotation I set in the values file.
Should be something like this:
```
external-dns.alpha.kubernetes.io/hostname: test-kafka-controller-0.dev.local
```

### What do you see instead?

.targetPod resolves to an empty string so that the annotation looks like this:

```
external-dns.alpha.kubernetes.io/hostname: .dev.local
```

### Additional information

The readme has an outdated config. In the section "Name resolution with External-DNS" it says to configure the annotation as `externalAccess.service.annotations` however according to the parameters section, it should be `externalAccess.controller.service.annotations`

**Comment by carrodher** [comment]
Thank you for bringing this issue to our attention. We appreciate your involvement! If you're interested in contributing a solution, we welcome you to create a pull request. The Bitnami team is excited to review your submission and offer feedback. You can find the contributing guidelines [here](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md).

Your contribution will greatly benefit the community. Feel free to reach out if you have any questions or need assistance.

**Comment by quang-streamkap** [comment]
Also having the same problem with versions "25.1.11"  and the latest "26.2.1". Please reopen this issue.

**Comment by carrodher** [comment]
Would you like to contribute by creating a pull request? The Bitnami team is excited to review your submission and offer feedback. You can find the contributing guidelines [here](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md).

Your contribution will greatly benefit the community. Feel free to reach out if you have any questions or need assistance.

**Comment by eli-gc** [comment]
No, I just had to use the default name.

**Comment by gscho** [comment]
This is a major annoyance for me - any idea how to fix it? I don't mind opening a PR but I don't understand why it doesn't work now as documented.

**Comment by gscho** [comment]
In case it helps anyone, I ended up working around the issue using loadBalancerAnnotations:

```
externalAccess:
    enabled: true
    autoDiscovery:
      enabled: true
    controller:
      service:
        type: LoadBalancer
        loadBalancerAnnotations:
          - service.beta.kubernetes.io/aws-load-balancer-name: "kafka-0"
            service.beta.kubernetes.io/aws-load-balancer-type: external
            service.beta.kubernetes.io/aws-load-balancer-scheme: "internal"
            service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: ip
            external-dns.alpha.kubernetes.io/hostname: "kafka-0.example.com"
          - service.beta.kubernetes.io/aws-load-balancer-name: "kafka-1"
            service.beta.kubernetes.io/aws-load-balancer-type: external
            service.beta.kubernetes.io/aws-load-balancer-scheme: "internal"
            service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: ip
            external-dns.alpha.kubernetes.io/hostname: "kafka-1.example.com"
          - service.beta.kubernetes.io/aws-load-balancer-name: "kafka-2"
            service.beta.kubernetes.io/aws-load-balancer-type: external
            service.beta.kubernetes.io/aws-load-balancer-scheme: "internal"
            service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: ip
            external-dns.alpha.kubernetes.io/hostname: "kafka-2.example.com"
```

**Comment by andresbono** [comment]
Hi, thank you for bringing this up again. `{{ .targetPod }}` resolution was originally implemented in #4832. Recovering the feature in this new PR: #24391. Once we merge the PR, this issue should be resolved.

Please, feel free to take a look at #24391.
